src/ripple/app/ledger/LedgerMaster.cpp

```diff
@@ -959,7 +947,7 @@ class LedgerMasterImp
                     WriteLog (lsTRACE, LedgerMaster) << "tryAdvance discovered missing " << missing;
                     if ((missing != RangeSet::absent) && (missing > 0) &&
                         shouldAcquire (mValidLedgerSeq, ledger_history_,
-                            ledger_history_index_, missing) &&
+                            getApp ().getSHAMapStore ().getCanDelete (), missing) &&
```

Comment:
Could this have been implemented in a way that does not introduce another loop in the cycle?